camviewer_ui_impl.py

```
@@ -1151,30 +1151,58 @@ def shutdown(self):
 
     def onfileSave(self):
         try:
-            fileName = QFileDialog.getSaveFileName(
+            filename = QFileDialog.getSaveFileName(
                 self,
                 "Save Image...",
-                self.cwd,
+                os.path.expanduser("~"),
                 "Images (*.npy *.jpg *.png *.bmp *.pgm *.tif)",
             )[0]
-            if fileName == "":
+            if filename == "":
                 raise Exception("No File Name Specified")
-            if fileName.lower().endswith(".npy"):
-                np.save(fileName, self.image)
+            if filename.lower().endswith(".npy"):
+                np.save(filename, self.image)
```

Just double checking, but can this fail silently? In the cases you check for if QImage.save fails (no write permission, file already exists, other), can this fail by throwing an exception which would not result in a retry? Also looking at https://github.com/numpy/numpy/blob/v2.1.0/numpy/lib/_npyio_impl.py#L507-L588
I think that this could override an existing file with the same name which shouldn't be possible in case someone tries to use the same filename for different images and ends up overwriting all but the last one.

---

You're right that I need to be more careful here. The exception being thrown would be "safe" but wouldn't result in a retry, giving inconsistent behavior.

I'll double-check about re-using the same name, but my fuzzy memory is that the file dialog gives you an "are you sure?" sort of error, so if we reach `np.save` the user has already confirmed that they want to replace the file.

---

Resolved by refactoring to make specifically `OSError` call the same method that we reach via the `QImage.save` checks.
Confirmed that the file select dialog warns the user sufficiently if they pick an already existing filename.